monai/bundle/scripts.py

```diff
@@ -822,6 +822,59 @@ def verify_metadata(
     logger.info("metadata is verified with no error.")
 
 
+def _get_net_io_info(
+    parser: ConfigParser | None = None, prefix: str = "_meta_#network_data_format", p: int = 1, n: int = 1, any: int = 1
+) -> tuple:
+    """
+    Get the input and output information defined in the metadata.
+
+    Args:
+        parser: a parser of the given bundle.
+        prefix: a prefix for the input and output id, which will be combined as `prefix#input` and
+            `prefix#output` to parse the input and output information in the `metadata.json` file of
+            bundle, default to `meta_#network_data_format`.
+        p: power factor to generate input data shape if dim of expected shape is "x**p", default to 1.
+        n: multiply factor to generate input data shape if dim of expected shape is "x*n", default to 1.
+        any: specified size to generate input data shape if dim of expected shape is "*", default to 1.
+
+    Returns:
+        input_channels: the channel number of the first input.
+        input_spatial_shape: the spatial shape of the first input.
+        input_dtype: the data type of the first input.
+        output_channels: the channel number of the first output.
+        output_dtype: the data type of the first output.
+        inputs_shape: a list contains the shape of every input.
+    """
+    if not isinstance(parser, ConfigParser):
+        raise AttributeError(f"'parser' should be a ConfigParser, got {type(parser)}.")
+
+    inputs_shape = []
+    prefix_key = f"{prefix}#inputs"
+    inputs_info_dict = parser.get(prefix_key)
+    for input_name in inputs_info_dict:
+        key = f"{prefix_key}#{input_name}#num_channels"
+        input_channels = parser.get(key)
+        key = f"{prefix_key}#{input_name}#spatial_shape"
+        input_spatial_shape = parser.get(key)
+        spatial_shape = _get_fake_spatial_shape(input_spatial_shape, p=p, n=n, any=any)
+        inputs_shape.append([1, input_channels, *spatial_shape])
```

Comment:
Here you use `for loop` to iterate a dictionary by keys, but save the results in a list, how to guarantee the order?
To be honest, I feel this logic is very confusing, can we remove all this logic and only support the "image" input for this first version?

Thanks.

Reply:
Ok, let's focus on one input for this first version.